apps/dev/mcp-renderer/mcp_renderer.py

```
@@ -74,13 +90,12 @@ def _send(self, msg: dict) -> None:
             self._stdin.flush()
 
     def _recv(self, timeout: float = 30.0) -> dict:
-        assert self._stdout is not None
         while True:
-            ready, _, _ = select.select([self._stdout], [], [], timeout)
-            if not ready:
+            try:
+                raw = self._stdout_q.get(timeout=timeout)
+            except queue.Empty:
                 raise TimeoutError(f"MCP server did not respond within {timeout:.0f}s")
```

Ruff (0.15.14)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)